facilitator/src/batch.rs

```diff
@@ -24,41 +26,50 @@ use uuid::Uuid;
 pub const AGGREGATION_DATE_FORMAT: &str = "%Y%m%d%H%M";
 
 /// Manages the paths to the different files in a batch
-pub struct Batch {
+pub struct Batch<H: Header, P: Packet> {
     header_path: String,
     signature_path: String,
     packet_file_path: String,
+
+    // These next two fields are not real and are used because not using H and P
+    // in the struct definition is an error.
+    phantom_header: PhantomData<*const H>,
+    phantom_packet: PhantomData<*const P>,
```

Comment:
nit: I'm pretty sure this works just as well:
```suggestion
    phantom_header: PhantomData<H>,
    phantom_packet: PhantomData<P>,
```
The only reason I had done `PhantomData<* const H>` is because I was copy-pasting from some example and didn't really understand what I was doing.